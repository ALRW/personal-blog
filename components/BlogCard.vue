<template>
  <div :class="blogClass">
    <nuxt-link :to="`articles/${post.path}`">
      <ProgressiveImage :image-path="post.image" />
      <h1 class="title">{{ post.title }}</h1>
      <h1 class="subtitle">{{ post.subtitle }}</h1>
    </nuxt-link>
  </div>
</template>

<script>
import ProgressiveImage from '@/components/ProgressiveImage'
export default {
  components: {
    ProgressiveImage
  },
  props: {
    post: {
      type: Object,
      default() {
        return {
          title: '',
          subtitle: '',
          id: '',
          path: '',
          image: ''
        }
      }
    },
    index: {
      type: Number,
      default: 0
    }
  },
  computed: {
    blogClass() {
      const size = this.index === 0 || this.index % 4 === 0 ? '12' : '4'
      return `column is-${size} blog-card link-pop`
    }
  }
}
</script>

<style>
.blog-card {
  padding: 1em 0 1em 0;
}
</style>
