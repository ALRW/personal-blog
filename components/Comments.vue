<template>
  <div class="comments">
    <button
      class="button is-fullwidth comment-button has-text-weight-semibold"
      @click="show = !show"
    >
      {{ show ? 'Hide Comments' : 'Show Comments' }}
    </button>
    <disqus
      v-if="show"
      :shortname="shortname"
      :identifier="identifier"
      :url="url"
      :title="title"
    ></disqus>
  </div>
</template>

<script>
import Disqus from 'vue-disqus/src/vue-disqus.vue'
export default {
  components: {
    Disqus
  },
  props: {
    identifier: {
      type: String,
      default: ''
    },
    title: {
      type: String,
      default: ''
    }
  },
  data() {
    return {
      show: false,
      shortname: 'andrew-werner',
      url: this.$route.query.page
    }
  }
}
</script>

<style>
.comments {
  border-bottom: 2px solid whitesmoke;
  padding: 2em 2em 2em 2em;
}
</style>
