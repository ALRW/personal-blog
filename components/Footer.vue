<template>
  <footer class="footer">
    <div class="columns is-multiline">
      <div class="column is-6">
        <p>Â© 2019 Andrew Werner. All rights reserved.</p>
      </div>
    </div>
  </footer>
</template>

<style>
.footer {
  background: white;
  border-top: 2px solid whitesmoke;
}
</style>
