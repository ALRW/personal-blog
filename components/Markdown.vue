<template>
  <div class="content">
    <div v-html="html"></div>
  </div>
</template>

<script>
export default {
  props: {
    html: {
      type: String,
      default: ''
    }
  }
}
</script>
