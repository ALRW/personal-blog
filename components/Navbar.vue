<template>
  <nav class="navbar has-shadow">
    <div class="navbar-brand">
      <nuxt-link class="navbar-item is-size-4" to="/">{{ title }}</nuxt-link>
      <a
        :class="{ 'is-active': isHamburgerActive }"
        role="button"
        class="navbar-burger"
        aria-label="menu"
        aria-expanded="false"
        @click="isHamburgerActive = !isHamburgerActive"
      >
        <span aria-hidden="true"></span>
        <span aria-hidden="true"></span>
        <span aria-hidden="true"></span>
      </a>
    </div>
    <div :class="{ 'is-active': isHamburgerActive }" class="navbar-menu">
      <div class="navbar-end">
        <nuxt-link class="navbar-item" to="/">{{ articles }}</nuxt-link>
        <nuxt-link class="navbar-item" to="/about">{{ about }}</nuxt-link>
      </div>
    </div>
  </nav>
</template>

<script>
export default {
  data() {
    return {
      title: `Andrew | Werner`,
      articles: `Articles`,
      about: `About Me`,
      isHamburgerActive: false
    }
  }
}
</script>
