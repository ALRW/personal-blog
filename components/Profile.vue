<template>
  <div>
    <figure class="image personal-profile">
      <img
        class="is-rounded"
        src="~assets/images/personal-profile.jpg"
        alt="personal profile"
      />
    </figure>
    <div class="has-text-centered">
      <h1 class="is-size-4">{{ title }}</h1>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    title: {
      type: String,
      default() {
        return ''
      }
    }
  }
}
</script>

<style>
.personal-profile {
  max-width: 300px;
  max-height: 300px;
}
.is-flex {
  justify-content: center;
}
.has-text-centered {
  margin-top: 5%;
}
</style>
