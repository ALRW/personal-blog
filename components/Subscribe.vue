<template>
  <div class="subscribe">
    <form @submit.prevent="handleSubmit">
      <div class="honeypot" aria-hidden="true">
        <input
          @input="handleHoney($event.target.value)"
          type="text"
          name="honey"
        />
      </div>
      <div v-if="success" class="has-text-centered subscribe-notification">
        <font-awesome-icon :icon="['far', 'check-circle']" size="6x" />
      </div>
      <div v-if="error" class="has-text-centered subscribe-notification">
        <font-awesome-icon :icon="['far', 'times-circle']" size="6x" />
      </div>
      <div class="field is-grouped">
        <div class="control is-expanded">
          <input
            @input="handleInput($event.target.value)"
            class="input"
            type="email"
            name="email"
            placeholder="my.name@something.com"
          />
        </div>
        <div class="control">
          <button class="button is-dark">Subscribe</button>
        </div>
      </div>
    </form>
  </div>
</template>

<script>
import jsonp from 'jsonp'
import queryString from 'query-string'
export default {
  data() {
    return {
      email: '',
      honeypot: false,
      success: false,
      error: false,
      loading: false,
      url: 'https://reasontree.us5.list-manage.com/subscribe/post-json',
      userId: '9b1673411a4fb84fdd7d90f79',
      listId: 'ee5a235ced'
    }
  },
  computed: {
    params() {
      return queryString.stringify({
        u: this.userId,
        id: this.listId,
        EMAIL: this.email
      })
    }
  },
  methods: {
    handleInput(value) {
      this.email = value.trim()
    },
    handleHoney(value) {
      this.honeypot = Boolean(value)
    },
    handleSubmit() {
      if (this.email.length === 0 || this.loading || this.honeypot) return
      this.success = false
      this.error = false
      this.loading = true
      const url = `${this.url}?${this.params}`
      jsonp(url, { param: 'c' }, this.handleResponse)
    },
    handleResponse(err, data) {
      this.loading = false
      if (err) {
        this.error = true
      } else {
        this.success = true
        this.email = ''
      }
    }
  }
}
</script>

<style>
.subscribe {
  border-top: 2px solid whitesmoke;
  border-bottom: 2px solid whitesmoke;
  padding: 2em 2em 2em 2em;
}
.subscribe-notification {
  margin-bottom: 2em;
}
.honeypot {
  position: absolute;
  left: -5000px;
}
</style>
