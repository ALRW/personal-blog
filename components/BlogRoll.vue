<template>
  <section class="section">
    <div class="columns is-multiline">
      <BlogCard
        v-for="(post, index) in posts"
        :key="index"
        :post="post"
        :index="index"
      />
    </div>
  </section>
</template>

<script>
import BlogCard from '@/components/BlogCard'
export default {
  components: {
    BlogCard
  },
  props: {
    posts: {
      type: Array,
      default() {
        return []
      }
    }
  }
}
</script>
