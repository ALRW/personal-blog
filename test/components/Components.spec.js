import { RouterLinkStub } from '@vue/test-utils'
import { subjectUnderTest } from '@/test/utils'
import BlogCard from '@/components/BlogCard'
import BlogRoll from '@/components/BlogRoll'
import Comments from '@/components/Comments'
import Footer from '@/components/Footer'
import Markdown from '@/components/Markdown'
import Navbar from '@/components/Navbar'
import Profile from '@/components/Profile'
import Subscribe from '@/components/Subscribe'

const mockPost = {
  title: 'title',
  subtitle: 'subtitle',
  id: 'id',
  path: '/path',
  image: '/path/to/image.jpg'
}

const components = [
  {
    name: 'BlogCard',
    component: BlogCard,
    options: {
      propsData: { post: mockPost, index: 0 },
      stubs: { NuxtLink: RouterLinkStub }
    }
  },
  {
    name: 'BlogRoll',
    component: BlogRoll,
    options: {
      propsData: { posts: [mockPost] }
    }
  },
  {
    name: 'Comments',
    component: Comments,
    options: {
      propsData: { identifier: 'identifier', title: 'title' },
      mocks: { $route: { query: { page: 'about' } } }
    }
  },
  {
    name: 'Footer',
    component: Footer,
    options: {}
  },
  {
    name: 'Markdown',
    component: Markdown,
    options: { propsData: { html: '<h1>Hello</h1>' } }
  },
  {
    name: 'Navbar',
    component: Navbar,
    options: { stubs: { NuxtLink: RouterLinkStub } }
  },
  {
    name: 'Profile',
    component: Profile,
    options: { propsData: { title: 'title', image: '/something.jpg' } }
  },
  {
    name: 'Subscribe',
    component: Subscribe,
    options: {}
  }
]

components.forEach(({ name, component, options }) => {
  describe(name, () => {
    it(`${name} is a vue component`, () => {
      expect(subjectUnderTest(component, options).isVueInstance()).toBeTruthy()
    })

    it(`${name} renders correctly`, () => {
      expect(subjectUnderTest(component, options).html()).toMatchSnapshot()
    })
  })
})
