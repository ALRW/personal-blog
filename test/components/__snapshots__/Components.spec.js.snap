// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogCard BlogCard renders correctly 1`] = `"<div class=\\"column is-12 blog-card link-pop\\"><a><progressiveimage-stub aspectratio=\\"is-2by1\\" imagepath=\\"/path/to/image.jpg\\"></progressiveimage-stub> <h1 class=\\"title\\">title</h1> <h1 class=\\"subtitle\\">subtitle</h1></a></div>"`;

exports[`BlogRoll BlogRoll renders correctly 1`] = `"<section class=\\"section\\"><div class=\\"columns is-multiline\\"><blogcard-stub post=\\"[object Object]\\" index=\\"0\\"></blogcard-stub></div></section>"`;

exports[`Comments Comments renders correctly 1`] = `
"<div class=\\"comments\\"><button class=\\"button is-fullwidth comment-button has-text-weight-semibold\\">
    Show Comments
  </button> <!----></div>"
`;

exports[`Footer Footer renders correctly 1`] = `"<footer class=\\"footer\\"><div class=\\"columns is-multiline\\"><div class=\\"column is-6\\"><p>Â© 2019 Andrew Werner. All rights reserved.</p></div></div></footer>"`;

exports[`Markdown Markdown renders correctly 1`] = `"<div class=\\"content\\"><div><h1>Hello</h1></div></div>"`;

exports[`Navbar Navbar renders correctly 1`] = `"<nav class=\\"navbar has-shadow\\"><div class=\\"navbar-brand\\"><a class=\\"navbar-item is-size-4\\">Andrew | Werner</a> <a role=\\"button\\" aria-label=\\"menu\\" aria-expanded=\\"false\\" class=\\"navbar-burger\\"><span aria-hidden=\\"true\\"></span> <span aria-hidden=\\"true\\"></span> <span aria-hidden=\\"true\\"></span></a></div> <div class=\\"navbar-menu\\"><div class=\\"navbar-end\\"><a class=\\"navbar-item\\">Articles</a> <a class=\\"navbar-item\\">About Me</a></div></div></nav>"`;

exports[`Profile Profile renders correctly 1`] = `"<div><progressiveimage-stub aspectratio=\\"is-square\\" imagepath=\\"/personal-profile.jpg\\" isrounded=\\"true\\"></progressiveimage-stub> <div class=\\"has-text-centered\\"><h1 class=\\"is-size-4\\">title</h1></div></div>"`;

exports[`ProgressiveImage ProgressiveImage renders correctly 1`] = `"<figure class=\\"image is-2by1\\"><div><img data-src=\\"[object Object]\\" data-loading=\\"[object Object]\\" class=\\"\\"></div></figure>"`;

exports[`SocialIcons SocialIcons renders correctly 1`] = `"<div class=\\"is-flex social-icons\\"><a href=\\"https://github.com/ALRW\\" target=\\"_blank\\" class=\\"link-pop\\"><font-awesome-icon-stub icon=\\"fab,github-square\\" size=\\"5x\\" class=\\"has-text-dark\\"></font-awesome-icon-stub></a> <a href=\\"https://www.linkedin.com/in/wernerandrew\\" target=\\"_blank\\" class=\\"link-pop\\"><font-awesome-icon-stub icon=\\"fab,linkedin\\" size=\\"5x\\" class=\\"has-text-dark\\"></font-awesome-icon-stub></a></div>"`;

exports[`Subscribe Subscribe renders correctly 1`] = `"<div class=\\"subscribe\\"><form><div aria-hidden=\\"true\\" class=\\"honeypot\\"><input type=\\"text\\" name=\\"honey\\"></div> <!----> <!----> <div class=\\"field is-grouped\\"><div class=\\"control is-expanded\\"><input type=\\"email\\" name=\\"email\\" placeholder=\\"my.name@something.com\\" class=\\"input\\"></div> <div class=\\"control\\"><button class=\\"button is-dark\\">Subscribe</button></div></div></form></div>"`;
